cmd/tea/gen_id.go

```diff
@@ -171,16 +241,29 @@ func extractIDs(filename string) (IDs, error) {
 			if !ok || !ts.Name.IsExported() {
 				continue
 			}
+			varName := ts.Name.Name
+			var typeName string
+			// Primitive type
 			typ, ok := ts.Type.(*ast.Ident)
-			if !ok {
-				continue
+			if ok {
+				typeName = typ.String()
 			}
-			t, ok := supportedType(typ.String())
-			if !ok {
-				fmt.Printf("Unsupported id: name=%s, type=%s\n", ts.Name.Name, typ.String())
+			// Composite type
+			compositeTyp, ok := ts.Type.(*ast.SelectorExpr)
```

Comment:
```suggestion
			compositeType, ok := ts.Type.(*ast.SelectorExpr)
```

Reply:
I've wanted to be consistent with `typ` at line 247 since type is the keyword